### PR TITLE
sdk: check systemd running for socket activation

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
ping @calavera @dave-tucker

**Comment by calavera** [comment]
As mentioned in https://github.com/docker/go-plugins-helpers/pull/29#issuecomment-194424467, socket activation should NOT be enabled without checking the host is using systemd.

File: sdk/handler.go [hunk 1]
```diff
@@ -81,3 +84,22 @@ func (h Handler) listenAndServe(proto, addr, group string) error {
 
 	return server.Serve(l)
 }
+
+func setupSocketActivation() (net.Listener, error) {
+	if !util.IsRunningSystemd() {
```
Comment (runcom) [inline sdk/handler.go]: @calavera I've added the check

File: sdk/tcp_listener.go [hunk 2]
```diff
@@ -14,10 +14,16 @@ const (
 )
 
 func newTCPListener(address string, pluginName string) (net.Listener, string, error) {
-	listener, err := sockets.NewTCPSocket(address, nil)
+	listener, err := setupSocketActivation()
```
Comment (calavera) [inline sdk/tcp_listener.go]: isn't this going to use a unix socket when you want a TCP one? it sounds like a bad behavior.

Comment (runcom) [inline sdk/tcp_listener.go]: Socket activation supports both tcp and Unix socket in unit files

Comment (runcom) [inline sdk/tcp_listener.go]: I mean the listen stream directive in the socket file can be 0.0.0.0:8080 for instance

Comment (runcom) [inline sdk/tcp_listener.go]: I'm going to recheck this but I'm pretty sure it worked correctly

**Comment by runcom** [comment]
my tests work well, I'm gonna research more about tcp - I've edited the PR to just add the check for systemd, @calavera PTAL

**Comment by calavera** [comment]
LGTM

**Comment by runcom** [comment]
ping @dave-tucker (who else? :))

**Pull request merged**